### ISSUE TITLE
[BUG] Split Strings Broke Code

### ISSUE THREAD

**Issue by BlackYuzia** [javascript-obfuscator/javascript-obfuscator#729]
## Info

I use **webpack** and **webpack plugin** for obfuscate.
Maybe this obfuscator broke webpack code... but I didn't sure on 100%...

Actualy, **need more test**. Maybe this trouble is not new and this is a duplicate...

## Description

Today, I was write some code. I found what my code, after obfuscation, didn't work.
I would think what this is trouble in properties rename (how I found in readme)
I will try change settings, disable all, enable and and etc.

After all tests, I found something...

If I setup ONLY property renamer - all is work fine.
If I setup ONLY split strings - all is ... fine?
But, if I setup settings:
```
      renameGlobals: true,
      renameProperties: true,
      splitStrings: true,
      splitStringsChunkLength: 1, // Actualy, I test 1 and 3 values. This both call error. If this value 5 or greater all is good
```

I get error (check stack trace) and my code is BROKEN now.

## Expected Behavior

Not broke code with too many symbols in **splitStringsChunkLength**

## Current Behavior

If splitStringsChunkLength < 5 (tested only on 3 and 1 values) I get broken code after obfuscation.

PS: Actually I didn't know what code really broke after that... maybe webpack, maybe code what I was write... 

## Steps to Reproduce
I was setup settings in to`splitStrings: true`,
1. Setup settings (look in description block)
2. Obfuscate code
3. Try start code
4. Maybe you get the same error like me

I think this trouble verry ... hard repeatfull.
I think without code (what I can't share currently) this trouble can be hard to repeat.

# Stack trace

```
VM2399:formatted:287 Uncaught TypeError: _0x1368a0[("b" + "q" + "C" + "a" + "s")] is not a function
    at <anonymous>:1:10081
    at <anonymous>:1:12128
    at <anonymous>:1:46619
```

## Minimal working example that will help to reproduce issue

I can't share with code from current project (maybe later ... if I would can do that)
Now, I can give only this line with error:
```
        if (_0x1368a0['b' + 'q' + 'C' + 'a' + 's'](_0x1368a0['P' + 'I' + 'n' + 'z' + 'c'], _0x1368a0['P' + 'I' + 'n' + 'z' + 'c'])) {
```

**PS: Sorry if I give too low info about all. Call and ask me if you want more info.**

**Comment by BlackYuzia** [comment]
@sanex3339 I get, how I think, the same error with `fetch`.

**Original Code**
```
fetch.get("https://google.com")
    .then(res => {
```
**Obfuscated Code**
```
 fetch[a0_0x4705('\x30\x78\x65\x30', '\x41\x40\x43\x45')]('https://google.com')[a0_0x4705('\x30\x78\x32\x65\x65', '\x75\x6f\x55\x35')](_0x3602ee=>{
```
**Stack Trace**
```
Uncaught TypeError: fetch[a0_0x4705(...)] is not a function
    at Object.<anonymous> (<anonymous>:1:270484)
    at _0x441d27 (<anonymous>:1:237215)
    at Object._0x29480.<computed> [as nAXpy] (<anonymous>:1:157168)
    at <anonymous>:1:247937
    at <anonymous>:1:247993
```

I found one more .... broke?
`fetch` is broke with `high` presset.

**Comment by BlackYuzia** [comment]
Hmm. It is kinda my mistake or something like this ...
Now I didn't get this error kinda ... :eyes:

If I approve what this is bug in this lib I would reopen this.

**Comment by BlackYuzia** [comment]
**Yep, I success fully test it now.**
Some code has been broken if I use `splitStringsChunkLength: 1`.
But, in currently moment, I didn't know what exactly code has been broken.

**Comment by sanex3339** [comment]
Please try to investigate it. I just re-checked that `splitStrings` option correctly works with emoji and everything looks ok.
So, maybe something in the code is using `function.toString()` for some reason (for example dependency injection in `angular 1`? 
![image](https://user-images.githubusercontent.com/9210534/91940817-c3dc5080-ed00-11ea-98ad-452ee2f4cce3.png)

**Comment by sanex3339** [comment]
Here is the test with `emoji` support 
https://github.com/javascript-obfuscator/javascript-obfuscator/commit/340b7283bed69388d6a7fe2563e9153efe14f48b#diff-2558a7cf4a8f074d417da29f3491ef9cR166

**Comment by sanex3339** [comment]
What obfuscator version are you using?

**Comment by BlackYuzia** [comment]
>  What obfuscator version are you using?
```
    "webpack": "^4.44.1",
    "webpack-cli": "^3.3.12",
    "webpack-obfuscator": "^1.12.0"
```

**Comment by sanex3339** [comment]
Ok, you've used the latest version. Now I need a reproducible example because right now I haven't an idea about the reason of this error. Try to use conditional comments to isolate parts of your code: https://github.com/javascript-obfuscator/javascript-obfuscator#conditional-comments

**Comment by BlackYuzia** [comment]
Okey I will try found broke code.
I answer later if I do that.

**Comment by BlackYuzia** [comment]
Success!

I found what has been broke!

Original code:
```
const obj = {a:0, b:1, c:2}; // Create object

console.log(obj.a)           // try get data
```

if length < 5, 
`console.log(obj.a)` was return `undefined`.
if length >= 5
`console.log(obj.a` was return `0`.

**Comment by BlackYuzia** [comment]
@sanex3339 this bug require `renameProperties`, but I think this is trouble by `splitStringsChunkLength`.
Because only if value of `splitStringsChunkLength` lower like 5, this bug is begun.

**Comment by sanex3339** [comment]
Cannot reproduce here:
https://obfuscator.io/

**Comment by sanex3339** [comment]
![image](https://user-images.githubusercontent.com/9210534/91963342-681cc200-ed16-11ea-9124-73acbd1178c2.png)

**Comment by BlackYuzia** [comment]
Maybe because `webpack`?

**Comment by BlackYuzia** [comment]
Yep, I also test the same code with https://obfuscator.io/ obfuscator.
**This bug did't reproduce.**

**Comment by BlackYuzia** [comment]
@sanex3339 if you can.
Try download `webpack`, `obfuscator` and try do the same.
But with `webpack`.

**Comment by sanex3339** [comment]
Cannot reproduce. 
Please create a repository with a full reproducible example.

**Comment by BlackYuzia** [comment]
@sanex3339 

> Cannot reproduce.
Please create a repository with a reproducible example, with configured webpack.

I will try later.
But, now check this pls. 

Value >= 5 code:
```
 _0x1f9cb7[a0_0x4faf('\x30\x78\x31\x37', '\x52\x5b\x72\x61')]), // success get value
// you can saw what this code have `object` and `function` what return right name of object key (if I right)
// _ox1f9cb7 is it object:
// _0x3c1aaa: 0
// _0x11c494: 1
// _0x50d40d: 2
// _0x2576e7: 3
```
Value < 5 code:
```
 _0x33c091['\x52' + '\x55' + '\x4e']  // fault get value, return undefined
// you can saw what this code have `object`, but key, what you can saw, didn't right. Obfuscator didn't change this key.
// _0x33c091 is it object:
// _0x3c1aaa: 0
// _0x11c494: 1
// _0x50d40d: 2
// _0x2576e7: 3
// note: I just copy this block of code and this object didn't the same (I mean obj keys) ... 
```

**Why this happen? I mean, if I setup value of length `5` all is good. If < 5, I get not right obj key.**
Why?

Note: I can't give original code right now. I try test all variations. Maybe I wrong somewhere...

**Comment by sanex3339** [comment]
The bug works only if you have object keys like `a`, `b` and `c` or it can be reproduced with keys like `foo`, `bar`, `baz`?

**Comment by sanex3339** [comment]
I still can't reproduce it, so I need a repository with this bug. If possible - with a small source codebase, only with the following code:
```
const obj = {a:0, b:1, c:2}; // Create object

console.log(obj.a)           // try get data
```

**Comment by BlackYuzia** [comment]
In main code obj key 3 char length and more.
But in clear code I can't reproduce that too. 

Maybe have any else requires for this bug?
I try test it currently.

**Comment by BlackYuzia** [comment]
@sanex3339 

**FUCK YEA! I DID IT!**

Yea, now I was know what code really destroy him self.
I test the same code with https://obfuscator.io/ but in this obfuscator all works fine.

~~Now I try to create a repo with setup build, settings and code.~~
https://github.com/BlackYuzia/obfuscator-bug-reply/tree/master

Code what broken from `webpack`
```
const KRUZYAISGAY = [0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2, 0, 1, 2];
const KRUZYANOTGAY = [0, 1, 2];
const KRUZYAISUNKNOW = [0, 1, 2, 3];

const KRUZYA = {
    KRUZYAISGAY: KRUZYAISGAY.length,
    KRUZYANOTGAY: KRUZYANOTGAY.length,
    KRUZYAISUNKNOW: KRUZYAISUNKNOW.length,
    KRUZYAISGAYNOTGAYUNKNOWN: KRUZYAISGAY.length - (KRUZYANOTGAY.length + KRUZYAISUNKNOW.length)
}

console.log("KRUZYA", KRUZYA, KRUZYA.KRUZYAISGAYNOTGAYUNKNOWN); // get undefined by "KRUZYA.KRUZYAISGAYNOTGAYUNKNOWN"
```

Note: Don't touch my Kruzya! :eyes:

**Comment by BlackYuzia** [comment]
Hint:
If I use `A` or `B` or `C` all is work fine.
If I use `AA` or `BB` or `CC` in object keys... all is broke.

**Comment by sanex3339** [comment]
Thank you! I'll check

**Comment by sanex3339** [comment]
Kruzya )

**Comment by sanex3339** [comment]
With the following code your example doesn't work too:
```
const object = {
    foobarbaz: 1
}

console.log(object.foobarbaz); // undefined
```

**Comment by sanex3339** [comment]
Obfuscated code looks like:
```
function (_0x3ba5ab, _0x4d6f18) {
        const _0x1d8bbf = { '_0x2dc854': 0x1 };
        console['log'](_0x1d8bbf['foobar' + 'baz']);
    }
```

So the property call of the object inside `console.log` is not renamed

**Comment by BlackYuzia** [comment]
> So the property call of the object inside console.log is not renamed

You think only inside of  `console.log(...)`?
I think if you use `anyFunction(...)` or just `if (...)` you get the same result.

**Comment by sanex3339** [comment]
No, not only inside `console.log`. I just keep it for the debug reasons.

**Comment by sanex3339** [comment]
Found the problem. It's not related to `webpack`. I moved `converting` transformers before `split-strings` transformers:
https://github.com/javascript-obfuscator/javascript-obfuscator/commit/9c04ededc7fda7b56cdfd1d4539af5dff1d89fe5#diff-d2e2508047c24e24bfd9bde17a6f0928R220

**Comment by sanex3339** [comment]
This will be fixed in `2.0.0` soon. Right now you can use `webpack-obfuscator@1.11.0`

**Comment by sanex3339** [comment]
PR: https://github.com/javascript-obfuscator/javascript-obfuscator/pull/731

**Comment by BlackYuzia** [comment]
**OFFTOP**
@sanex3339 would you know how I can setup `webpack-obfuscator` plugin to run before other plugins?
I setup up one more plugin AFTER `webpack-obfuscator`. But I get obfuscated code with changes from last plugin too.

I need use a loader?

I want to create a `header` before `code line` but I can't do that.
I try use `banner-webpack-plugin` but `webpack-obfuscator` was obfuscate this banner too :eyes:

**Comment by sanex3339** [comment]
`banner-webpack-plugin` adds comment?

**Comment by BlackYuzia** [comment]
yep. (But can input `raw` text too. Like `console.log("some text");`)

**Comment by sanex3339** [comment]
If it adds comment, add `@preserve` or `@license` line to this comment and it won't be removed after obfuscation.
For code wrap this code with conditional comments https://github.com/javascript-obfuscator/javascript-obfuscator#conditional-comments

**Comment by BlackYuzia** [comment]
Thx. I will try.

**Comment by sanex3339** [comment]
I've just released both `javascript-obfuscator@1.12.1` and `webpack-obfuscator@1.12.1`
